govcd/lbservicemonitor.go

```diff
@@ -191,28 +195,32 @@ func validateCreateLBServiceMonitor(lbMonitorConfig *types.LBMonitor) error {
 	return nil
 }
 
-func validateReadLBServiceMonitor(lbMonitorConfig *types.LBMonitor) error {
+func validateGetLbServiceMonitor(lbMonitorConfig *types.LbMonitor, egw *EdgeGateway) error {
+	if !egw.HasAdvancedNetworking() {
+		return fmt.Errorf("only advanced edge gateways support load balancers")
+	}
+
 	if lbMonitorConfig.ID == "" && lbMonitorConfig.Name == "" {
 		return fmt.Errorf("to read load balancer service monitor at least one of `ID`, `Name` fields must be specified")
 	}
 
 	return nil
 }
 
-func validateUpdateLBServiceMonitor(lbMonitorConfig *types.LBMonitor) error {
+func validateUpdateLbServiceMonitor(lbMonitorConfig *types.LbMonitor, egw *EdgeGateway) error {
```

Comment:
Should the EdgeGateway be the receiver as opposed to parameter for these few `validate` refactored functions?

Reply:
I thought of it. Can change. My point was to avoid further polluting EdgeGateway with functions which don't help being used separately.